gelic-idealab/komodo-unity
# Release/v0.5.8

# v0.5.8 Release: Interactable Drawings, Refactored DrawingInstanceManager

## Interactable Drawings

Please see https://github.com/gelic-idealab/komodo-unity/pull/100 

## Refactoring

Renamed Draw_Refresh to ReceiveDrawUpdate and refactored into the following functions:

* ContinueLine
* EndLine
* DeleteLine
* HideLine
* ShowLine

Combined lineRenderersInQueue and allStrokeIDValidator into a single dictionary, lineRendererFromID, and made sure any uses of that dictionary went through new functions that talk directly to the ID. Did not change the overall functionality.

De-coupled DrawingInstanceManager from NetworkObjectsManager.

Added various warnings for edge cases, like trying to move a line that exists in your client but not in someone else's (because you drew it before they joined).

The motivation is that we should now be able to more easily debug the content. In particular, we can address the warning in the linked pull request and display the drawing line / stroke ID. 

### Type of change


### How Has This Been Tested?

Test basic single-client transforming of drawings

* Connect with a client, then...
* make drawings
* try to transform your own drawings (translate / rotate / scale)

Test basic synchronized transforming of drawings

* Connect with two clients, then...
* make drawings
* try to transform your own drawings (translate / rotate / scale)
    * expected result: transform synchronizes correctly
* try to transform each others' drawings (translate / rotate / scale)
    * expected result: transform is successful
    * expected result: transform synchronizes correctly

Note that undo only applies to the draw action and erase action, and not to any transform actions.



[Duplicate all of the above for each distinct feature / bug fix / etc.]
_____

# Test Configuration

* Firefox
* Oculus Quest 1
* Oculus Quest 2
- [x] PC VR (Link / AirLink)

# Checklist:


Modified from this article:
Phillip Johnston, “A GitHub Pull Request Template for Your Projects - Embedded Artistry,” Embedded Artistry, Aug. 04, 2017. https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ (accessed Jul. 22, 2021).
